servatrice/src/servatrice_database_interface.cpp

```diff
@@ -98,22 +98,8 @@ bool Servatrice_DatabaseInterface::execSqlQuery(QSqlQuery *query)
 
 bool Servatrice_DatabaseInterface::usernameIsValid(const QString &user)
 {
-    QString result;
-    result.reserve(user.size());
-    foreach (const QChar& c, user) {
-        switch (c.category()) {
-        case QChar::Letter_Uppercase:       //[A-Z]
-        case QChar::Letter_Lowercase:       //[a-z]
-        case QChar::Number_DecimalDigit:    //[0-9]
-        case QChar::Punctuation_Connector:  //[_]
-        case QChar::Punctuation_Dash:       //[-]
-            result += c;
-        default: 
-            break;
-        }
-    }
-    result = result.trimmed();
-    return (result.size() == user.size());
+    static QRegExp re = QRegExp("[a-zA-Z0-9_\.-]+");
```

Comment:
Update: double escaping the dot seems to fix the warning https://travis-ci.org/Cockatrice/Cockatrice/jobs/58828986#L849 : 
```
[ 80%] Building CXX object cockatrice/CMakeFiles/cockatrice.dir/src/dlg_filter_games.cpp.o
/home/travis/build/Cockatrice/Cockatrice/servatrice/src/servatrice_database_interface.cpp:101:45: warning: 
      unknown escape sequence '\.'
    static QRegExp re = QRegExp("[a-zA-Z0-9_\.-]+");
                                            ^~
1 warning generated.
```